chain/types/ethtypes/eth_legacy_155_transactions.go

```diff
@@ -0,0 +1,291 @@
+package ethtypes
+
+import (
+	"fmt"
+
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/go-state-types/big"
+	typescrypto "github.com/filecoin-project/go-state-types/crypto"
+
+	"github.com/filecoin-project/lotus/build"
+	"github.com/filecoin-project/lotus/chain/types"
+)
+
+var _ EthTransaction = (*EthLegacy155TxArgs)(nil)
+
+// EthLegacy155TxArgs is a legacy Ethereum transaction that uses the EIP-155 chain replay protection mechanism
+// by incorporating the chainId in the signature.
+// See how the `V` value in the signature is derived from the chainId at
+// https://github.com/ethereum/go-ethereum/blob/86a1f0c39494c8f5caddf6bd9fbddd4bdfa944fd/core/types/transaction_signing.go#L424
+// For EthLegacy155TxArgs, the digest that is used to create a signed transaction includes the `ChainID` but the serialised RLP transaction
+// does not include the `ChainID` as an explicit field. Instead, the `ChainID` is included in the V value of the signature as mentioned above.
+type EthLegacy155TxArgs struct {
+	LegacyTx *EthLegacyHomesteadTxArgs
```

Comment:
As written, this is a guarantee to cause bugs in the future when people accidentally call the "inner" methods on `EthLegacyHomesteadTxArgs` instead of the correct methods (on `EthLegacy155TxArgs`). Please rewrite this, you could:

- create an inner struct that's just the fields (no logic), and then have two separate structs that contain that struct
- just duplicate the fields in `EthLegacy155TxArgs`, it's fine to do so
- if you really want to keep the existing pattern, make `LegacyTx` private, and rewrite external usages of it to access it properly (eg. by creating a constructor)

Reply:
Fixed this by making it private.